llvm/include/llvm/ADT/iterator.h

```diff
@@ -267,6 +267,8 @@ class iterator_adaptor_base
     return *static_cast<DerivedT *>(this);
   }
   using BaseT::operator-;
+  template <bool Enabled = bool(BaseT::IsRandomAccess),
+            std::enable_if_t<Enabled, int> = 0>
```

Comment:
```suggestion
  template <bool Enabled = BaseT::IsRandomAccess,
            typename = std::enable_if_t<Enabled>>
```
I think this is a little bit more idiomatic